src/systems/waypoint-system.js

```diff
@@ -323,7 +329,24 @@ export class WaypointSystem {
       this.waitOneTick = false;
       return;
     }
+
+    const hashUpdated = window.location.hash !== "" && this.previousWaypointHash !== window.location.hash;
+
+    if (hashUpdated && this.initialSpawnHappened) {
+      this.previousWaypointHash = window.location.hash;
+      const spawnPointName = window.location.hash.replace("#", "");
+      const spawnPoint = this.ready.find(c => c.el.className === spawnPointName);
+      if (spawnPoint) {
+        this.occupySpawnPoint(spawnPoint);
+        this.currentMoveToSpawn = null;
+        this.currentMoveToSpawnResolve = null;
+        this.nextMoveToSpawn = null;
+        this.nextMoveToSpawnResolve = null;
```

Comment:
It used to be that we had code that looked like,
```js
waypointSystem.moveToSpawnPoint().then(()=> { // do something after spawning })
```
We don't do that anymore (https://github.com/mozilla/hubs//blob/master/src/hub.js#L440), but still I'm nervous that `null`ifying these book-keeping variables may cause unintended consequences down the road if ever that changes. I don't think you actually need to modify these. 

If you want to make sure to avoid conflicts with a spawn-flow-in-progress (which happens over several frames), maybe add `this.currentMoveToSpawn === null && this.nextMoveToSpawn === null` to the check above: 
```js
if (hashUpdated && this.initialSpawnHappened && this.currentMoveToSpawn === null && this.nextMoveToSpawn === null)
```

Reply:
With the current flow it won't conflict as the initial spawn will happen and I'll just resolve the hashtag later. I have removed all the nulls anyway as they were not affecting it